src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentReferenceCaptureLoweringPass.cs

```diff
@@ -33,32 +35,106 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
         {
-            // If we can't get a nonempty attribute name, do nothing because there will
-            // already be a diagnostic for empty values
-            var identifierToken = DetermineIdentifierToken(node);
-            if (identifierToken == null)
-            {
-                return node;
-            }
-
             // Determine whether this is an element capture or a component capture, and
             // if applicable the type name that will appear in the resulting capture code
             var componentTagHelper = (parent as ComponentIntermediateNode)?.Component;
             if (componentTagHelper != null)
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken, componentTagHelper.GetTypeName());
+                return new ReferenceCaptureIntermediateNode(identifier, componentTagHelper.GetTypeName());
             }
             else
             {
-                return new ReferenceCaptureIntermediateNode(identifierToken);
+                return new ReferenceCaptureIntermediateNode(identifier);
+            }
+        }
+
+        private bool ShouldGenerateField(IntermediateNode parent)
+        {
+            var parameters = parent.FindDescendantNodes<TagHelperDirectiveAttributeParameterIntermediateNode>();
+            for (var j = 0; j < parameters.Count; j++)
+            {
+                var parameter = parameters[j];
+                if (parameter.TagHelper.IsRefTagHelper() && parameter.BoundAttributeParameter.Name == "suppressField")
+                {
+                    if (parameter.HasDiagnostics)
+                    {
+                        parent.Diagnostics.AddRange(parameter.GetAllDiagnostics());
+                    }
+
+                    parent.Children.Remove(parameter);
+
+                    if (parameter.AttributeStructure == AttributeStructure.Minimized)
+                    {
+                        return false;
+                    }
+
+                    // We do not support non-minimized attributes here because we can't allow the value to be dynamic.
+                    // As a design/experience decision, we don't let you write @ref:suppressField="false" even though
+                    // we could parse it. The rationale is that it's misleading, you type something that looks like code,
+                    // but it's not really.
+                    parent.Diagnostics.Add(ComponentDiagnosticFactory.Create_RefSuppressFieldNotMinimized(parameter.Source));
+                }
+            }
+
+            if (parent is ComponentIntermediateNode component && component.Component.IsGenericTypedComponent())
+            {
+                // We cannot automatically generate a 'ref' field for generic components because we don't know
+                // how to write the type.
+                parent.Diagnostics.Add(ComponentDiagnosticFactory.Create_RefSuppressFieldRequiredForGeneric(parent.Source));
```

Comment:
Would probably make sense to return false here to avoid flooding the user with extraneous errors due to trying to generate a generic field.

Reply:
Yes, you're right, I meant to do that.